ospd_openvas/openvas_db.py

```diff
@@ -60,219 +44,234 @@
-    """
-    global DB_ADDRESS
-    try:
-        result = subprocess.check_output(['openvassd', '-s'],
-                                         stderr=subprocess.STDOUT)
-        result = result.decode('ascii')
-    except OSError:
-        # the command is not available
-        return 2
-
-    if result is None:
-        return 2
-
-    path = None
-    for conf in result.split('\n'):
-        if conf.find("db_address") == 0:
-            path = conf.split('=')
-            break
-
-    if path is None:
-        return 2
-
-    DB_ADDRESS = str.strip(path[1])
-
-
-def max_db_index():
-    """Set the number of databases have been configured into kbr struct.
-    """
-    global MAX_DBINDEX
-    try:
-        ctx = kb_connect()
-        resp = ctx.config_get("databases")
-    except redis.RedisError:
-        return 2
-
-    if isinstance(resp, dict) is False:
-        return 2
-    if len(resp) == 1:
-        MAX_DBINDEX = int(resp["databases"])
-    else:
-        print("Redis: unexpected reply length %d" % len(resp))
-        return 2
-
-def set_global_redisctx(ctx):
-    """ Set the global set the global REDISCONTEXT.
-    """
-    global REDISCONTEXT
-    REDISCONTEXT = ctx
-
-def db_init():
-    """ Set DB_ADDRESS and max_db_index. """
-    if get_db_connection() or max_db_index():
-        return False
-    return True
-
-def try_database_index(ctx, i):
-    """ Check if it is already in use. If not set it as in use and return.
-    """
-    try:
-        resp = ctx.hsetnx(DBINDEX_NAME, i, 1)
-    except:
-        return 2
-
-    if isinstance(resp, int) is False:
-        return 2
-
-    if resp == 1:
-        return 1
-
-def kb_connect(dbnum=0):
-    """ Connect to redis to the given database or to the default db 0 .
-    """
-    global DB_INDEX
-    if get_db_connection() is 2:
-        return 2
-    try:
-        ctx = redis.Redis(unix_socket_path=DB_ADDRESS,
-                          db=dbnum,
-                          socket_timeout=SOCKET_TIMEOUT, charset="latin-1",
-                          decode_responses=True)
-    except ConnectionError as e:
-        return {"error": str(e)}
-    DB_INDEX = dbnum
-    return ctx
-
-def db_find(patt):
-    """ Search a pattern inside all kbs. When find it return it.
-    """
-    for i in range(0, MAX_DBINDEX):
-        ctx = kb_connect (i)
-        if ctx.keys(patt):
-            return ctx
-
-def kb_new():
-    """ Return a new kb context to an empty kb.
-    """
-    ctx = db_find(DBINDEX_NAME)
-    for index in range(1, MAX_DBINDEX):
-            if try_database_index(ctx, index) == 1:
-                ctx = kb_connect(index)
+
+class OpenvasDB(object):
+    """ Class to connect to redis, to perform queries, and to move
+    from a KB to another."""
+
+    def __init__(self):
+        # Path to the Redis socket.
+        self.db_address = ""
+
+        # Name of the namespace usage bitmap in redis.
+        self.dbindex_name = "GVM.__GlobalDBIndex"
```

Comment:
If it is a constant variable and not changed in this class or a child class it should be a class variable instead.